test/lit/passes/gufa-refs.wast

```diff
@@ -361,29 +361,30 @@
     )
   )
 
-  ;; CHECK:      (func $nondeterminism (type $none_=>_funcref) (result funcref)
-  ;; CHECK-NEXT:  (block $label$1 (result funcref)
-  ;; CHECK-NEXT:   (drop
-  ;; CHECK-NEXT:    (br_on_func $label$1
-  ;; CHECK-NEXT:     (i31.new
-  ;; CHECK-NEXT:      (i32.const 1337)
+  ;; CHECK:      (func $nondeterminism (type $none_=>_dataref) (result dataref)
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (block $label$1 (result dataref)
+  ;; CHECK-NEXT:    (drop
+  ;; CHECK-NEXT:     (br_on_data $label$1
+  ;; CHECK-NEXT:      (i31.new
+  ;; CHECK-NEXT:       (i32.const 1337)
+  ;; CHECK-NEXT:      )
   ;; CHECK-NEXT:     )
   ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:    (ref.null none)
   ;; CHECK-NEXT:   )
-  ;; CHECK-NEXT:   (ref.null nofunc)
   ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (ref.null none)
   ;; CHECK-NEXT: )
-  (func $nondeterminism (result funcref)
+  (func $nondeterminism (result dataref)
     ;; This block is sent an i31 and a null. The null is compatible with the
     ;; type and the i31 is not. The order in which we process this matters:
     ;;
     ;;  * If the i31 arrives first, we'll filter it out as incompatible with the
     ;;    type of the block. Then the null arrives and the final result is that
     ;;    null, which we can then optimize the block to return.
     ;;  * Or, if the null arrives first, then when the i31 arrives the
-    ;;    combination of nullfunc + i31 is Many (since the types are
-    ;;    incompatible). We then filter that to the block's type, ending up with
-    ;;    a cone of funcref. We cannot optimize in that case, unlike before.
+    ;;    combination of none + i31 is ??????
```

Comment:
I think I updated the test so it's still testing what it is supposed to, but I'm not sure how to update this text here since the types are now compatible. Can you tell me if the test looks good and what this text should say?

Reply:
If the types are now compatible then I think this is testing something else. Can the `br_on_func` that gets an `i31` be replaced with a `br_on_cast` that receives an incompatible type?

Reply:
We could do e.g. `(br_on_cast $label i31 (struct.new ...))`, but there's no longer any way to combine types in completely different hierarchies, which is what we had before and is what created the `Many` mentioned in the comment.

Reply:
I see. I guess this test has not been entirely testing the same thing then, maybe since the null changes actually.

I think we don't need it in that case. Let's delete it - it tested something that can't be expressed now.

Reply:
Sounds good, thanks!